### DISCUSSION TITLE
Why deprecate passing arguments with an environmental variable

### DISCUSSION THREAD

**Discussion by jmstrupp-psjh** [github/gh-gei#737] [Q&A] [answered]
In version 0.29
- Added --aws-bucket-name to gh gei generate-script **and removed --azure-storage-connection-string**

It seems odd to me that you would deprecate an arguement you can pass on the command line in favor of only an environmental variable.  This Breaks previous code and provides no real benefit - why not allow both (with the command line taking precedent)?

**Comment by ArinGhazarian** [comment, marked as answer]
Hi @jmstrupp-psjh 👋,

We didn't remove the arg from the `migrate-repo` command, we just removed it from the `generate-script` command mainly because of security reasons so that the secret won't appear in the generated script. This is similar to `--github-source-pat` and `--github-traget-pat` args which are not supported in `generate-script` command again for security reasons.

Now since the `migrate-repo` command still accepts the `--azure-storage-connection-string` as an option, it is possible to edit the generated script and manually add it to all `migrate-repo` commands.

**Reply by timrogers** [reply]
Hi @jmstrupp-psjh 👋🏻 

Just on top of @ArinGhazarian's answer, I wanted to apologise for any confusion or pain caused here. We definitely could have done a better job of communicating the reasons behind this change and explaining how users can adapt to it.

If you have any questions, please don't hesitate to reach out.
